backend/src/email/middlewares/email-template.middleware.ts

```diff
@@ -379,10 +387,114 @@ const uploadProtectedCompleteHandler = async (
   }
 }
 
+/**
+ * Downloads the file from the vault url and checks that its columns match the attributes provided in the template.
+ * If a template has not yet been uploaded, do not write to the message logs, but prompt the user to upload a template first.
+ * If the template and csv do not match, prompt the user to upload a new file.
+ * @param req
+ * @param res
+ * @param next
+ */
+const tesseractHandler = async (
```

Comment:
Just a thought, given that the logic is fairly similar to the normal `uploadCompleteHandler`, I was wondering if there's a way we can refactor this to reduce duplication in the code. Perhaps we can centralise the logic in `UploadService`.